### ISSUE TITLE
Monitor Localization Accuracy using Intersection Over Union test while training .

### ISSUE THREAD

**Issue by stavBodik** [keras-team/keras#8669]
Hi , I have written code for saving events to TensorBoard file in order to monitor and test localization accuracy while training.

I am using Keras backend, there is option to make custom function and pass it while compiling the model. But the problem is that its hard to debug it because its runs on the GPU .
https://keras.io/metrics/

So Actually I decided not to use this approach and call my own function directly for making averages on each batch end , which predicts and makes IOU test.
Then when epoch ends ,I use on_epoch_end function of my callbacks to write the final average log files.

In order to perform the test I am using y_true array( which is all the prior default boxes then I filter from it only the ground truth positive boxes) and images for the corresponding epoch.

But unfortunately this array is encoded while the assign boxes step and when I am trying to decode it I am getting garbage, for each bounding box the result is an vector which is [ 1, 0, 0, 0 ].

The decode,encode functions I am using are taken from ssd_utils.py , class BBoxUtility .

I attach my test function and the utils code, may some one look at tell how can I fix it please ?

[IOU_test.zip](https://github.com/fchollet/keras/files/1524620/IOU_test.zip)

Thanks!
